app/models/event.rb

```diff
@@ -0,0 +1,13 @@
+class Event < ApplicationRecord
+  belongs_to :creator, class_name: 'User'
+
+  has_many :event_attendances, foreign_key: :attended_event_id, dependent: :destroy
+  has_many :attendees, through: :event_attendances, source: :attendee
+
+  validates :name, :event_date, presence: true
+
+  delegate :name, to: :creator, prefix: true
+
+  scope :past, -> { where('event_date <= ?', Time.zone.now) }
```

Comment:
- [ ] past should only look for events already past, and upcoming should check for events occurring right now and in the future.